examples/sessions/src/main.rs

```diff
@@ -124,3 +135,18 @@ impl UserId {
         Self(Uuid::new_v4())
     }
 }
+
+fn parse_cookie(cookie: String) -> HashMap<String, String> {
+    let mut map = HashMap::new();
+    let v: Vec<&str> = cookie.split(";").collect();
+    for cookie in &v {
+        let kv: Vec<&str> = cookie.split("=").collect();
+        if kv.len() < 2 {
+            return HashMap::new();
+        }
+        let key = kv[0].trim_start();
+        let value = kv[1].trim_end();
+        map.insert(key.to_string(), value.to_string());
+    }
+    map
+}
```

Comment:
Do you know if `cookie` has a built in solution to this? Perhaps using [`CookieJar`](https://docs.rs/cookie/0.15.1/cookie/struct.CookieJar.html). Having to parse cookies yourself doesn't feel something you should be doing.

Reply:
Sorry for the lack of explanation. For example, if we run the program with a cookie of `csrf=xxx', it will panic as follows.
````
2021-11-07T13:37:06.702658Z DEBUG example_sessions: listening on 127.0.0.1:3000
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Invalid byte 61, offset 4.', examples_sessions\src\main. rs:102:79
````

So I tried to deal with the problem by parsing the cookie, creating a `HashMap` with the key and value of the cookie, and passing the value of the cookie given by this program to `load_session`.

> Do you know if cookie has a built in solution to this? Perhaps using CookieJar. Having to parse cookies yourself doesn't feel something you should be doing.

I only knew about the `cookie` crate😢 Yes, it's not something I would do.